plugins/player.js

```diff
@@ -21,26 +26,29 @@ module.exports = function(pluto) {
             console.log("getting song urls");
             songs[song.id] = songs[song.id] || {ignore: []};
             muzik.getLink(song, songs[song.id].ignore, function(name,url) {
-                var downloadCommand = "curl \"" + url.shellEscape()+"\" > " + songURL;
-                exec(downloadCommand, {async: true}, function(code, output) {
-                    if (code == 0) {
-                        console.log("Downloaded file");
-                        exec(playCommand, {async:true},function(code,output){
-                            if (code == 0) {
-                                console.log("playing");
-                            } else {
-                                songs[song.id].ignore.push(url);
-                                pluto.saveStorage("songs");
-                                pluto.emitEvent("music::play", song);
-                            }
-                        });
-                    } else {
-                        console.log("error trying another link");
+                request.head(url, function(err, res, body) {
+                    if (err || res.headers['content-type'].indexOf("audio") == -1 || res.headers['content-length']/1000000 < MIN_MB) {
                         songs[song.id].ignore.push(url);
                         pluto.saveStorage("songs");
                         pluto.emitEvent("music::play", song);
+                    } else {
+                        downloading.song = song;
+                        request(url).pipe(fs.createWriteStream(songURL)).on('close', function() {
```

Comment:
Going to get progress from this https://github.com/IndigoUnited/node-request-progress